File: ws/src/main/scala/de/thm/ii/fbs/model/Participant.scala
package de.thm.ii.fbs.model

import org.json.JSONObject

/**
  * Course participant
 *
  * @param user The participant
  * @param role The participants role in the course
  * @param visible visibility state
  * @author Andrej Sajenko
  */
case class Participant(user: User, role: CourseRole.Value, visible: Boolean = true) {
  /**
    * visibility state of User in Conference component
    */
  var isVisible: Boolean = visible
  /**
    * Calls underlying users toJson function
    * @return user toJson
    */
  val toJson: JSONObject = user.toJson().put("courseRole", role)
}

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/model/v2/Participant.kt
package de.thm.ii.fbs.model.v2

import com.fasterxml.jackson.annotation.JsonProperty
import de.thm.ii.fbs.model.v2.security.User

data class Participant(@JsonProperty val user: User, @JsonProperty val role: CourseRole, @JsonProperty val visible: Boolean = true)
